go/libraries/doltcore/env/dolt_docs.go

```diff
@@ -42,25 +44,27 @@ func LoadDocs(fs filesys.ReadWriteFS) (*Docs, error) {
 				return nil, err
 			}
 			val.NewerText = data
+			docsWithCurrentText[i] = val
 		}
 	}
-	return &docsWithCurrentText, nil
+	return docsWithCurrentText, nil
 }
 
-func CreateDocs(fs filesys.ReadWriteFS) (*Docs, error) {
+func CreateDocs(fs filesys.ReadWriteFS) (Docs, error) {
 	docs := *AllValidDocDetails
-	for _, doc := range docs {
+	for i, doc := range docs {
 		doc.NewerText = getInitialDocText(doc.DocPk)
+		docs[i] = doc
 	}
 	err := docs.Save(fs)
 	if err != nil {
 		return nil, err
 	}
-	return &docs, nil
+	return docs, nil
 }
 
-func (docs *Docs) Save(fs filesys.ReadWriteFS) error {
-	for _, doc := range *docs {
+func (docs Docs) Save(fs filesys.ReadWriteFS) error {
```

Comment:
You're calling this as an error recovery mechanism in a lot of places, but not checking the error value when doing so. This is a problem, but probably not a big one. Couple suggestions for dealing with it:
1) Do less of this kind of error recovery. It's being applied very broadly in some cases, i.e. writing docs on any error in a command, not just ones that might have touched docs
2) Check the error condition from this method, and return a new error that wraps the one that triggered the recovery save
3) If nothing else, use log a warning.

The absolute worst thing we could do would be to corrupt the user's files and not even tell them we did so, which is possible with the current version of this code.

Reply:
Makes sense. I tried to get rid of it everywhere where it was an unnecessary extra step. Now you should only see deliberate Saves, with immediate error handling.